python/ray/tests/test_gcs_fault_tolerance.py

```diff
@@ -28,14 +29,20 @@ def increase(x):
     indirect=True)
 def test_gcs_server_restart(ray_start_regular):
     actor1 = Increase.remote()
-    result = ray.get(actor1.method.remote(1))
-    assert result == 3
+    for i in range(0, 3):
+        result = ray.get(actor1.method.remote(1))
+        assert result == 3
 
     ray.worker._global_node.kill_gcs_server()
     ray.worker._global_node.start_gcs_server()
 
-    result = ray.get(actor1.method.remote(7))
-    assert result == 9
+    # Sleep for 5 seconds to ensure that the
+    # GCS client gets data after the GCS server restarts.
+    time.sleep(5)
```

Comment:
Why don't we keep calling ray.actors and wait using that instead of sleep here?

Reply:
Yes, let's not rely on sleeps. How about just something to make sure that the old GCS server PID has exited?

Reply:
This is not to wait for the GCS server to restart successfully, but to wait for the GCS client to retrieve the GCS server data and reconnect the actor. If we keep calling ray.actors and wait using that instead of sleep here, it will execute successfully because it may not have fetched data from the GCS server and connect the actor again. This is why previous use cases have been successful.

Reply:
I will add a UT instead of modifying the python integration testcase.

Reply:
I will add a UT instead of modifying the python integration testcase.

Reply:
Can you tell us which part of code in 'direct_actor_transport_test.cc' verified that the duplicate notification is ignored?